toast-tk-runtime/src/main/java/io/toast/tk/runtime/TestRunner.java

```diff
@@ -86,19 +96,60 @@ private void enrichTestBlockResults(ITestPage testPage, TestBlock testBlock) {
 		testBlock.setTestFailureNumber(nbBlockFailures);
 
 	}
+	
+	public void kill() {
+		this.interupted = true;
+		if(blockRunner != null) {
+			if(blockRunner instanceof TestBlockRunner) {
+				((TestBlockRunner) blockRunner).kill();
+			}
+		}
+	}
 
 	private void runTestPageBlocks(final ITestPage testPage) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)